web/portal/client/cypress/e2e/Calendar/HolidayDate/HolidayDate.cy.js

```diff
@@ -0,0 +1,168 @@
+      .should('eq', 201);
+  });
+
+  ///////////////////////////////
+  // PUT
+  ///////////////////////////////
+  it('edit HolidayDate', () => {
+    cy.intercept('GET', '**/api/client/holiday_dates/1', {
+      ...HolidayDateItem,
+    }).as('getHolidayDate-1');
+
+    cy.usePactIntercept(
+      {
+        method: 'PUT',
+        url: `**/api/client/holiday_dates/${EditHolidayDate.response.body.id}`,
+        response: EditHolidayDate.response,
+      },
+      'EditHolidayDate'
+    );
+
+    cy.get('svg[data-testid="EditIcon"]').first().click();
+    const { name, eventDate, locution } = EditHolidayDate.request;
+    cy.fillTheForm({
+      name,
+      eventDate,
+      locution,
+    });
+    cy.contains('Holiday dates');
+    cy.usePactWait(['EditHolidayDate'])
+      .its('response.statusCode')
+      .should('eq', 200);
+  });
+
+  ///////////////////////
+  // DELETE
+  ///////////////////////
+  it('delete HolidayDate', () => {
+    cy.intercept('DELETE', '**/api/client/holiday_dates/*', {
+      statusCode: 204,
+    }).as('deleteHolidayDate');
+
+    cy.get('td button > svg[data-testid="DeleteIcon"]').first().click();
+    cy.contains('Remove element');
+    cy.get('div.MuiDialog-container button')
+      .filter(':visible')
+      .contains('Yes, delete it')
+      .click();
+
+    cy.get('header').should('contain', 'Holiday dates');
+
+    cy.usePactWait(['deleteHolidayDate'])
+      .its('response.statusCode')
+      .should('eq', 204);
+  });
+
+  ///////////////////////
+  // POST
+  ///////////////////////
+  it('add HolidayDateRange', () => {
+    cy.usePactIntercept(
+      {
+        method: 'POST',
+        url: '**/api/client/holiday_dates_range*',
+        response: NewHolidayDateRange.response,
+        matchingRules: NewHolidayDateRange.matchingRules,
+      },
+      'createHolidayDateRange'
+    );
+
+    cy.get('button svg[data-testid="MoreHorizIcon"]').first().click();
+
+    cy.contains('Add Holiday date range').click();
+
+    const { startDate, endDate, locution, name } = NewHolidayDateRange.request;
+    cy.fillTheForm({
+      startDate,
+      endDate,
+      locution,
+      name,
+    });
+
+    cy.get('header').should('contain', 'Holiday dates');
+    cy.usePactWait('createHolidayDateRange')
+      .its('response.statusCode')
+      .should('eq', 201);
+  });
+
+  ///////////////////////
+  // POST
+  ///////////////////////
+  it('import CSV', () => {
+    cy.intercept('POST', '**/api/client/holiday_dates/mass_import*', {
+      statusCode: 201,
+      body: {
+        success: true,
+        errorMsg: '',
+        failed: 0,
+      },
+      headers: {
+        'Content-Type': 'application/json; charset=utf-8',
+      },
+    }).as('createImportCSV');
+
+    cy.get('button svg[data-testid="MoreHorizIcon"]').first().click();
+    cy.contains('Import CSV').click();
+    cy.get('h2').should('contain', 'Import Holiday Dates');
+
+    cy.fixture('HolidayDate/massImport.csv').then((fileContent) => {
```

Comment:
custom command like this:
```suggestion
Cypress.Commands.add('uploadFile', (filePath, selector) => {
  cy.fixture(filePath).then((fileContent) => {
    cy.get(selector).then((input) => {
      const blob = new Blob([fileContent], { type: 'text/csv' });
      const file = new File([blob], 'massImport.csv', { type: 'text/csv' });
      const dataTransfer = new DataTransfer();
      dataTransfer.items.add(file);
      input[0].files = dataTransfer.files;
      cy.wrap(input).trigger('change', { force: true });
    });
  });
});
```

Reply:
Then you can use it in this file like this:
```suggestion
 cy.uploadFile('HolidayDate/massImport.csv', 'input[type="file"]');
```